### PR TITLE
Add Detection for CVE-2024-10542

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
image

### CODE REVIEW

File: tests/exploits/cve_2024_10542_test.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import requests_mock as req_mock
+
+from agent import definitions
+from agent.exploits import cve_2024_10542
+
+
+def testCVE202410542_whenVulnerable_reportFinding(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """CVE-2024-10781 unit test: case when target is vulnerable."""
```
Comment (deadly-panda) [inline tests/exploits/cve_2024_10542_test.py:12]:
```suggestion
    """CVE-2024-10542 unit test: case when target is vulnerable."""
```

File: tests/exploits/cve_2024_10542_test.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    )
+    assert vulnerability.technical_detail == (
+        "http://localhost:80 is vulnerable to CVE-2024-10542, CleanTalk Plugin Arbitrary Plugin Installation (Via Reverse DNS Spoofing)"
+    )
+
+
+def testCVE202410542_whenSafe_reportNothing(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """CVE-2024-10781 unit test: case when target is safe."""
```
Comment (deadly-panda) [inline tests/exploits/cve_2024_10542_test.py:48]:
```suggestion
    """CVE-2024-10542 unit test: case when target is safe."""
```

File: tests/exploits/cve_2024_10542_test.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    assert accept is True
+    assert len(vulnerabilities) > 0
+    vulnerability = vulnerabilities[0]
+    assert (
+        vulnerability.entry.title
+        == "CleanTalk Plugin Arbitrary Plugin Installation (Via Reverse DNS Spoofing)"
+    )
+    assert vulnerability.technical_detail == (
+        "http://localhost:80 is vulnerable to CVE-2024-10542, CleanTalk Plugin Arbitrary Plugin Installation (Via Reverse DNS Spoofing)"
+    )
```
Comment (deadly-panda) [inline tests/exploits/cve_2024_10542_test.py:42]: Add an assert on the risk rating please.

File: tests/exploits/cve_2024_10542_test.py (hunk 1 above)
Comment (nmasdoufi-ol) [reply tests/exploits/cve_2024_10542_test.py]: Resolved.

**Review by adnaneserrar — APPROVED** [review]

**Review by ybadaoui-ostorlab — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**